opencog/reasoning/pln/rules/evaluation-to-member.scm

```diff
@@ -0,0 +1,75 @@
+; =============================================================================
+; GeneralEvaluationToMemberRule 
+;	Takes EvaluationLinks with >=2 Arguments and creates a Member Link
+;
+;	EvaluationLink (pred D (ListLink B C))
+;		becomes
+;	MemberLink( B (SatisfyingSetLink( Variable $X 
+;		(EvaluationLink (pred D (ListLink X C))))))
+;						&
+;	MemberLink( B (LambdaLink( Variable $X 
+;		(EvaluationLink (pred D (ListLink X C))))))
+; -----------------------------------------------------------------------------
```

Comment:
You could decompose this rule into

*1) Evaluation2MemberRule
```
EvaluationLink
   D
   L
|-
MemberLink
   L
   SatisfyingSet D
```

*2.a) BinaryPredicateFirstArgProjectionRule

```
PredicateNode D         -- such that D is binary
|-
LambdaLink
    X
    AverageLink
        Y
        EvaluationLink
            D
            ListLink X Y
```

*2.b) BinaryPredicateSecondArgProjectionRule

```
PredicateNode D         -- such that D is binary
|-
LambdaLink
    Y
    AverageLink
        X
        EvaluationLink
            D
            ListLink X Y
```

etc, for n-ari predicate, just to circumvent the lack of functional type checker for now.

Of course I do understand that maybe those multiple rules may bloat inference chains more, and then inference control must take care of not turning any n-ari predicates into unary ones.